lib/gtm.js

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+
+/**
+ * Google tag manager related utility functions
+ */
+
+/**
+ * @param  {...any} args - data to feed to dataLayer.push
+ */
+export function pushToDataLayer(...args) {
+  if (typeof window === 'undefined') return; // Skip in SSR
+
+  if (!window.dataLayer) window.dataLayer = [];
+  window.dataLayer.push(...args);
+}
+
+/**
+ * @param {object} props - data to send to pushDataLayer on component mount
+ */
+export function PushToDataLayer({ ...props }) {
```

Comment:
Is it a good way to declare react effect function with the same name in capitalization?

Reply:
Hmm I invented this, not seeing this pattern somewhere else.

I wrap the function as a functional component because I feel that it blends well with apollo-client's hooks. It will look more complicated if we implement this logic in useEffect hooks because we will need to duplicate the if-else logic regarding data loading flag inside the hook.

Reply:
Yeah, this fp will be helpful, but I'm not sure the naming is good.
how about name it as `usePushToDataLayer`?

Reply:
But this is a component, not a hook.

Reply:
When a React component integrated with apollo-client, it looks like this:

```javascript
function Component() {
  const {data, loading, error} = useQuery(...) ;
  if(loading) {
    return <>...</>
  }
  if(error) {
    return <>...</>
  }

  return <>...</>
}
```

When `PushToDataLayer` is implemented as a component, we can just put it in the last return statement:
```javascript
function Component() {
  const {data, loading, error} = useQuery(...) ;
  if(loading) {
    return <>...</>
  }
  if(error) {
    return <>...</>
  }

  return <>...<PushToDataLayer ...></>
}
```

But if we implement it as hooks, we will need to duplicate the if statement:

```javascript
function Component() {
  const {data, loading, error} = useQuery(...) ;
  usePushToDataLayer(loading, error);

  if(loading) {
    return <>...</>
  }
  if(error) {
    return <>...</>
  }

  return <>...</>
}

function usePushToDataLayer(loading, error) {
  if(loading || error) return;
  useEffect(() => {
    pushToDataLayer(...);
  }, [])
}
```

Reply:
Both ways are okay for me. 
if using functional component -> I think that naming might be need to changed. 
if using custom effect -> I think that duplicated if-else is not a problem. Because of that if-else give this function more cohesion which is like the concept __AOP__

```js
function usePushToDataLayer(enable) {
  if(!enable) return;
  useEffect(() => {
    pushToDataLayer(...);
  }, [])
}
```

Reply:
Updated.